File: backend/src/main/java/com/bookbook/domain/chat/dto/ChatRoomResponse.java
package com.bookbook.domain.chat.dto;

import com.bookbook.domain.chat.entity.ChatRoom;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
public class ChatRoomResponse {
    
    private Integer id;
    private String roomId;
    private Integer rentId;
    private String bookTitle;
    private String bookImage;
    private Integer otherUserId;
    private String otherUserNickname;
    private String otherUserProfileImage;
    private String lastMessage;
    
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime lastMessageTime;
    
    private Long unreadCount;
    private boolean isActive;
    
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdDate;
    
    public static ChatRoomResponse from(ChatRoom chatRoom, 
                                      String bookTitle, 
                                      String bookImage,
                                      String otherUserNickname, 
                                      String otherUserProfileImage,
                                      Long unreadCount) {
        return ChatRoomResponse.builder()
                .id(chatRoom.getId())
                .roomId(chatRoom.getRoomId())
                .rentId(chatRoom.getRentId())
                .bookTitle(bookTitle)
                .bookImage(bookImage)
                .otherUserId(null) // 컨트롤러에서 설정
                .otherUserNickname(otherUserNickname)
                .otherUserProfileImage(otherUserProfileImage)
                .lastMessage(chatRoom.getLastMessage())
                .lastMessageTime(chatRoom.getLastMessageTime())
                .unreadCount(unreadCount)
                .isActive(chatRoom.isActive())
                .createdDate(chatRoom.getCreatedDate())
                .build();
    }
}

File: backend/src/main/java/com/bookbook/domain/chat/dto/ChatRoomResponse.kt
package com.bookbook.domain.chat.dto

import com.bookbook.domain.chat.entity.ChatRoom
import com.fasterxml.jackson.annotation.JsonFormat
import java.time.LocalDateTime

data class ChatRoomResponse(
    val id: Long,
    val roomId: String,
    val rentId: Long,
    val bookTitle: String?,
    val bookImage: String?,
    val otherUserId: Long?,
    val otherUserNickname: String?,
    val otherUserProfileImage: String?,
    val lastMessage: String?,

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val lastMessageTime: LocalDateTime?,

    val unreadCount: Long,
    val isActive: Boolean,

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    val createdDate: LocalDateTime
) {
    companion object {
        fun from(
            chatRoom: ChatRoom,
            bookTitle: String?,
            bookImage: String?,
            otherUserNickname: String?,
            otherUserProfileImage: String?,
            unreadCount: Long
        ): ChatRoomResponse {
            return ChatRoomResponse(
                id = chatRoom.id,
                roomId = chatRoom.roomId,
                rentId = chatRoom.rentId,
                bookTitle = bookTitle,
                bookImage = bookImage,
                otherUserId = null, // 컨트롤러에서 설정
                otherUserNickname = otherUserNickname,
                otherUserProfileImage = otherUserProfileImage,
                lastMessage = chatRoom.lastMessage,
                lastMessageTime = chatRoom.lastMessageTime,
                unreadCount = unreadCount,
                isActive = chatRoom.isActive,
                createdDate = chatRoom.createdDate
            )
        }
    }
}